### PR TITLE
Add folding guns

### PR DESCRIPTION
**Pull request by NoahGoldFox**
#### Summary

SUMMARY: content "add several folding guns"

#### Purpose of change
I saw that there were no folding guns in the game, and that folding guns would be very convenient, especially with the new nested inventory and "longest side" making it much harder to carry guns around. These folding guns add realistic but somewhat rare loot that are really useful, especially to survivors who pack light.

#### Describe the solution

adding in new folding guns and putting them in item-groups

#### Describe alternatives you've considered
Adding a tool kit that can take down any rifle, but thats not very realistic.

#### Testing

Spawn in each gun, fold and unfold each one several times. Fold with gun attachments and while reloaded, try shooting while fold to make sure it cant. Unfold in backpack and make sure it falls out of backpack since it got too long.

#### Additional context
This PR adds the : 
[ruger 10/22 takdown](https://ruger.com/products/1022Takedown/models.html)
[chiappa little badger](https://www.chiappafirearms.com/family.php?id=26)
[Ruger PC carbine](https://ruger.com/products/pcCarbine/models.html)
[Keltec su-16](https://www.keltecweapons.com/firearms/rifles/su16/)
[foldAR](https://foldar.com/product/double-foldar15-pistol-9-5-56-nato-2020-gen3/)
[chiappa alaskan takedown 44](https://www.chiappafirearms.com/product.php?id=124)
[chiappa wildlands 45-70](https://www.chiappafirearms.com/product.php?id=646)

I was going to add more guns, like the [Paratus folding battle rifle](https://drdtactical.com/paratus/) and the [m96 windrunner takedown 12.7mm](http://www.edmarms.com/products/m96.htm) among others, but i decided not to since those were slightly too rare.

### CODE REVIEW

File: data/json/items/gun/22.json [hunk 1]
```diff
@@ -379,5 +379,129 @@
...
       }
     ]
+  },
+  {
+    "id": "chiappabadger",
+    "looks_like": "rifle_22",
+    "type": "GUN",
+    "reload_noise_volume": 10,
+    "name": { "str_sp": "Chiappa Little Badger" },
+    "description": "A lightweight, ultra-compact, break-action folding rifle.  It is extremely basic, with a simple wire stock (which includes little 5 shot ammo holder) and a plain barrel.  Use to fold ",
```
Comment (Qrox) [inline data/json/items/gun/22.json:389]:
```suggestion
    "description": "A lightweight, ultra-compact, break-action folding rifle.  It is extremely basic, with a simple wire stock (which includes little 5 shot ammo holder) and a plain barrel.  Use to fold.",
```

File: data/json/items/gun/22.json [hunk 2]
```diff
@@ -379,5 +379,129 @@
...
+    "flags": [ "NEVER_JAMS", "RELOAD_EJECT" ],
+    "pocket_data": [ { "pocket_type": "MAGAZINE", "ammo_restriction": { "22": 1 } } ],
+    "use_action": { "target": "foldedchiappabadger", "msg": "You fold the smol rifle", "menu_text": "fold rifle", "type": "transform" }
+  },
+  {
+    "id": "foldedchiappabadger",
+    "type": "GUN",
+    "category": "weapons",
+    "name": { "str": "folded chiappa little badger" },
+    "description": "A folded chiappa badger 22LR rifle.  It is very compact in this form.  Use to unfold",
```
Comment (Qrox) [inline data/json/items/gun/22.json:427]:
```suggestion
    "description": "A folded chiappa badger 22LR rifle.  It is very compact in this form.  Use to unfold.",
```

File: data/json/items/gun/22.json [hunk 3]
```diff
@@ -379,5 +379,129 @@
...
+    "color": "brown",
+    "use_action": { "target": "chiappabadger", "msg": "You unfold the smol rifle", "menu_text": "unfold rifle", "type": "transform" }
+  },
+  {
+    "id": "ruger_1022takedown",
+    "looks_like": "ruger_1022",
+    "type": "GUN",
+    "reload_noise_volume": 10,
+    "name": { "str_sp": "Ruger 10/22 takedown" },
+    "description": "A ruger 10/22 version with the ability to be easily taken down into 2 parts for easier carrying.  It has polymer furniture and is also slightly lighter then the normal 10/22.  ",
```
Comment (Qrox) [inline data/json/items/gun/22.json:445]:
```suggestion
    "description": "A ruger 10/22 version with the ability to be easily taken down into 2 parts for easier carrying.  It has polymer furniture and is also slightly lighter then the normal 10/22.",
```
No need for spaces at the end of description.

**Comment by Hirmuolio** [comment]
The foldable weapons could probably shoot while folded. Just with really bad accuracy.

Also it would be a bit weird to have both folding stock weapon mod and separate folding weapons.

**Comment by NoahGoldFox** [comment]
> The foldable weapons could probably shoot while folded. Just with really bad accuracy.
> 
> Also it would be a bit weird to have both folding stock weapon mod and separate folding weapons.

Actually of the guns this adds only the su-16 can fire at all while folded, and the version i added has a stock which makes that harder.

**Comment by dissociativity** [comment]
you might want to make sure they allow less attachments than a standard rifle to prevent awkward folding stock further reducing volume exploits if you haven't already.

**Comment by NoahGoldFox** [comment]
> you might want to make sure they allow less attachments than a standard rifle to prevent awkward folding stock further reducing volume exploits if you haven't already.

ah yee. i removed the stock mount from the foldar since its folding stock is part of its foldiness.

**Review by kevingranade — CHANGES_REQUESTED** [review]
I'm not sure if you're aware, but the primary holdup is you have a bunch of load errors.

**Comment by NoahGoldFox** [comment]
I will fix those in a while, its not abandoned.

**Comment by NoahGoldFox** [comment]
Im going to come back to this sometime later, but for now im closing it.

**Pull request closed without merging**